### DISCUSSION TITLE
Questions about sources

### DISCUSSION THREAD

**Discussion by jonnyhotchkiss** [microsoft/winget-cli#3880] [General] [answered]
Hi, I've used winget for a while, but still struggle to use as my single package manager.

1. There's apps that I use that I can get with winget, but that doesn't get the latest version - can the sources be contributed to, so the latest known version is obtainable? Or does it depend on the source and it's owner, and the type of resource?
Example: `winget install sonic-pi` will get SonicPi (v4, but not latest 4.5)
2. There's apps I want to get from places like github - is it possible to add github as a source?! 
3. On sources, the store seems to be flaky/unpredictable: sometimes I can get something even when Im not using a msstore account, sometimes I can only get after a login, sometimes even then the install fails... is this the best place to discuss msstore-winget-source usage queries?
4. What about extended functionality like installing assets from user-communities, eg scripts, like ReaPack does for Reaper? Is this something that's already possible with winget, or has anything similar been implemented, or are there better solutions to managing things like community-scripts ?

Thanks!

**Comment by denelon** [comment, marked as answer]
We welcome community contribution to the default source "winget" over at https://github.com/microsoft/winget-pkgs.

There are several packages pointing to releases at GitHub. As long as those are "official" sources, they are fine.

The "msstore" source is managed by the Microsoft Store team. WinGet will try to use a "device" based entitlement if no user is logged into the Microsoft Store application.

WinGet wasn't designed to distribute scripts. The PowerShell Gallery might be a better mechanism for that.
